Set integer ruler intervals when zoomed in

hydrobeam opened the pull request:
<!-- Please reference any relevant issue number below, optionally with a "Closes"/"Resolves"/"Fixes" prefix -->

## Summary

Closes #1927 

Adds a check for when the interval would be too small and then applies `.ceil()` to make it a whole number. Now, the progression of intervals is:

...200, 100, 50, 25, 12.5,7(6.25), 4(3.125), 2(...),1

## Discussion

I chose to start rounding at 12.5 since 12.5 is a fairly legible number. It only started looking weird below that. A slight issue is that 7 is used when 6.25 is rounded up, which isn't a clean number for the ruler :shrug:. I couldn't think of a clean way to get it to 8 or 6 without messing up the other numbers (i thought about something like: `ruler_interval += ruler_interval % 2` to force evenness, but then `ruler_interval` bottoms out at 2, which felt too big). 

## Results

When toggling the node graph display, the ruler now looks like this:
![image](https://github.com/user-attachments/assets/2f0d903b-d2e6-41fb-a43c-669683da1727)

And for very high zoom percentages (> 60000%), the ruler tops out at an interval of 1. 
![image](https://github.com/user-attachments/assets/99b75cf7-7861-4706-947f-fbd775d22dec)

Previously, the long decimals would overlap each other and look bad:
![image](https://github.com/user-attachments/assets/f0300704-614a-437d-b9a8-dbef1a8e0740)

Keavon commented:
!build

Keavon commented:
This is great! Nice work on implementing it in such a simple equation.

You're right that it's a bit weird for the progression to jump from increments of 12.5 to 7 to 4. Can you add a special case to turn the progression from the first line into the second line here?:
```
...200, 100, 50, 25, 12.5, 7 (6.25), 4 (3.125), 2 (1.5625), 1
...200, 100, 50, 25, 12.5, 6 (6.25), 4 (3.125), 2 (1.5625), 1
```

I'm also thinking that it would be great if, at some point when you zoom in more, we start to get the decimals for `.0`, `.1`, `.2`, ..., `.9` and continuing that at additional orders of magnitude (`.01`, `.001`, etc.). But that doesn't necessarily have to prevent this PR from merging, as it's already very useful in its current form. Would you like to take a crack at that though?

hydrobeam commented:
I wasn't sure about going lower than 1.0 before since going below 1 unit means you're already insanely zoomed in, but I've added it in [74f6647](https://github.com/GraphiteEditor/Graphite/pull/1966/commits/74f6647a47b50397e7599fcee2c1e1e02f3d2a77), also modified the progression to go to 6 and not 7.

Showcase of sub 1.0 scales:

https://github.com/user-attachments/assets/b3a506b7-cdc7-41a9-ab61-9a19a585cc97

The max zoom% seems to be 1million%, so you can't really go infinitely deep.

Also, I needed to add the `.toFixed()` stuff in `RulerInput.svelte` because there were some weird floating point shenanigans going on that I couldn't resolve:

![image](https://github.com/user-attachments/assets/d21b8f30-a248-42bb-93f9-3181569631bb)

So I opted to truncate the strings depending on how many decimal slots were expected.

Keavon commented:
!build

Keavon commented:
This is perhaps a separate issue but I'm bringing it up as a concern, even if it's not something you fix in this PR:

If the labeled increments aren't divisible by 5, we end up forcing sub-increments divided by 5 and 10 anyways, causing a misalignment. This is captured at 1500x document zoom and we see it's choosing to display the labels in increments of 6. Each gray bar is set at 1px increments in document space. We can see how they misalign because of the 6 bars between each labeled number, but only 5 minor (10 micro) increments.

![capture](https://github.com/user-attachments/assets/71a1ae95-3a0d-41e5-b2bb-11f8ea5d915f)

I'm open to suggestions about the right solution here, but I think it would basically mean either forcing the labeled numbers to always be increments of 5, or changing the number of minor and micro increments to divide evenly into the distances between the labeled increments.

Keavon reviewed (approved):
Thank you for this feature improvement! If you'd like to follow this up with the fix to the issue I mentioned above, you could mention the same issue so both this and that PR would be the ones that close the same issue (I don't think filing a separate issue makes sense). I'll also direct your attention to #1482 and #1969 as other ruler-related improvements that you might be interested in working on later. Cheers!
